app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -0,0 +1,135 @@
+      unless validate_metric(metric)
+        raise MetricValidationError, "Validation error: metric must be defined"
+      end
+      starts = (start_time - interval).to_i.in_milliseconds
+      ends = end_time.to_i.in_milliseconds
+      bucket_duration = "#{interval}s"
+      metrics = collect_metrics(metric[:id], metric[:type], starts, ends, bucket_duration)
+      process_data(metric, metrics)
+    end
+
+    def first_and_last_capture(metric)
+      unless validate_metric(metric)
+        raise MetricValidationError, "Validation error: metric must be defined"
+      end
+      case metric[:type]
+      when "GAUGE" then
+        min = @ems.metrics_connect.gauges.get_data(metric[:id], starts: 0, limit: 1, order: 'ASC')
+        max = @ems.metrics_connect.gauges.get_data(metric[:id], starts: 0, limit: 1, order: 'DESC')
+      when "COUNTER" then
+        min = @ems.metrics_connect.counters.get_data(metric[:id], starts: 0, limit: 1, order: 'ASC')
+        max = @ems.metrics_connect.counters.get_data(metric[:id], starts: 0, limit: 1, order: 'DESC')
+      when "AVAILABILITY" then
+        min = @ems.metrics_connect.avail.get_data(metric[:id], starts: 0, limit: 1, order: 'ASC')
+        max = @ems.metrics_connect.avail.get_data(metric[:id], starts: 0, limit: 1, order: 'DESC')
+      else raise MetricValidationError, "Validation error: unknown type #{metric_type}"
+      end
+      [min[0]['timestamp'], max[0]['timestamp']]
+    end
+
+    def collect_metrics(metric_id, metric_type, starts, ends, bucket_duration)
+      case metric_type
+      when "GAUGE" then gauges(metric_id, starts, ends, bucket_duration)
+      when "COUNTER" then counters(metric_id, starts, ends, bucket_duration)
+      when "AVAILABILITY" then availabilities(metric_id, starts, ends, bucket_duration)
+      else raise MetricValidationError, "Validation error: unknown type #{metric_type}"
+      end
+    end
+
+    def gauges(metric_id, starts, ends, bucket_duration)
+      sort_and_normalize(
+        @ems.metrics_connect.gauges.get_data(metric_id, :starts => starts, :ends => ends,
+                                             :bucketDuration => bucket_duration))
+    end
+
+    def counters(metric_id, starts, ends, bucket_duration)
+      compute_derivates(
+        sort_and_normalize(
+          @ems.metrics_connect.counters.get_data(metric_id, :starts => starts, :ends => ends,
+                                                 :bucketDuration => bucket_duration)))
+    end
+
+    def availabilities(metric_id, starts, ends, bucket_duration)
+      sort_and_normalize(
+        @ems.metrics_connect.avail.get_data(metric_id, :starts => starts, :ends => ends,
+                                            :bucketDuration => bucket_duration))
+    end
+
+    def sort_and_normalize(data)
+      # Sorting and removing last entry because always incomplete
+      # as it's still in progress.
+      norm_data = (data.sort_by { |x| x['start'] }).slice(0..-2)
```

Comment:
NaN values are discarded inside sort_and_normalize method.
Values are strings for availability type of metrics and integers for counters and gauges.
Not sure if it is ok to maintain this or we should add specific strong typing on that.
Still experimenting on the handling part of the metrics.